### ISSUE TITLE
Adding jail / proot support

### ISSUE THREAD

**Issue by dirkcgrunwald** [trampgeek/jobe#3]
This is perhaps not an issue so much as a query - it would be useful to have a chroot'd jail for the applications, but this is typically a pain to implement.

For another system where I execute arbitrary code from students (our computer systems class), I recently retrofitted the server to use "proot" ( http://proot.me/ ). This turned out to be very easy.

The setup code was something like this (create /tmp/jailXYZZY directory)
		    $jail = tempdir( CLEANUP => 1 );
		    system("/bin/mkdir $jail/bin") == 0
			or log_die("unable to create proot $jail/bin");
		    system("/bin/mkdir $jail/tmp") == 0
			or log_die("unable to create proot $jail/tmp");
		    system("/bin/cp $BINDIR/bufbomb $jail/bin") == 0
			or log_die("unable to copy bufbomb to proot jail");

this makes /tmp/jail/tmp, and /tmp/jail/bin. The binary is then coped it i/tmp/jail/bin

and then the execution is something like:

proot -r $jail $prootlibs bufbomb 

where prootlibs is $PROOT_LIBS="-b /etc/ld.so.cache -b /lib/ld-linux.so.2 -b /lib32/libc.so.6 -w /bin ";
in other words, the files that should be mapped into the proot jail.

My situation was much simpler because I only needed specific files mapped in (the ones above), but proot can also handle standard jail tasks w/o sudo.

**Comment by trampgeek** [comment]
Thanks for the heads-up on `proot`, Dirk. It sounds very interesting. I've never felt quite strong enough to implement the full chroot jail because, as you say, it's a pain to have to set up all the language tools and execution environments within the jail. But his looks promising -- I'll investigate it in the next break, but I'll close the issue for now.

**Comment by sophia00w** [comment]
Adding jail and proot support would be a useful improvement for users who need more flexibility in restricted environments. It could make the setup process smoother and expand compatibility across different use cases. Clear documentation and examples would also help users get started quickly. For those needing related legal references, https://okaloosacountycourt.org may provide additional information worth reviewing.
